### PR TITLE
feat: add typescript typings

### PR DESCRIPTION
**Pull request by lholmquist**
This adds typesctript types based on the JSDoc

### CODE REVIEW

**Review by mhdawson — APPROVED** [review]
LGTM

**Review by helio-frota — APPROVED** [review]

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+    token?: string;
+    username?: string;
+    password?: string;
+    apiServer?: string;
+    server?: string;
+    insecure?: string;
+    forceLogin?: string;
+}): Promise<object>;
+/**
+  The login function will login
```
Comment (danbev) [inline index.d.ts:26]: Should this be `logout` perhaps?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+
+  @param {object} [options] - Options object for the deploy function
```
Comment (danbev) [inline index.d.ts:4]: Should this be options for the `login` function?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+    password?: string;
+    apiServer?: string;
+    server?: string;
+    insecure?: string;
+    forceLogin?: string;
+}): Promise<object>;
+/**
+  The login function will login
+
+  @param {object} [options] - Options object for the deploy function
```
Comment (danbev) [inline index.d.ts:28]: Should this be options for the `logout` function instead of for the `deploy` function?

**Review by danbev — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 2 above)
Comment (lholmquist) [reply index.d.ts:4]: :facepalm:

**Review by danbev — APPROVED** [review]

**Review by danbev — APPROVED** [review]

**Pull request merged**